### ISSUE TITLE
Revised process tree concept

### ISSUE THREAD

**Issue by egmontkob-work** [firebuild/firebuild#566]
There was something that never felt right to me with the tree containing a mixture of two node types (ExecedProcess vs. ForkedProcess). I just couldn't explain what it really is :) Let alone those pointers that make "long jumps", up to the next exec ancestor...

Now that we've gained a lot of experience, and now that we understand that exec points are the primary essential things we care about due to the project's very nature (those are the only places where we have a chance to cache and shortcut), and forks are just a secondary technical necessity (to track different fd mappings, separate socket connections etc.) ...

I think that the following would be a nicer model:

The top-level ProcessTree would actually be an ExecTree or ExecPointTree, whatever we call it. Each edge would represent a successful execve(). A node could have multiple children because of internal forks.

_Within_ each such node, there'd be one ForkTree with at least one node, tracking all the stuff that we need to track differently across forks (pid, fds, connection fd etc.).

In other words, the current flat tree would be split into two levels: from a bird's eye view we'd seen the execs only, and if we zoom in then there'd be a second level about the forks. "Long jumps" would become simple edges in the outer level.

I think this model would be mentally better, and maybe help us always unconsciously know if a word like "inherited" cares about forks or not.

---

I understand that it's quite some work (mostly boring mechanical "smart search & replace") for little benefits, and our current model is good enough. I'm leaving this idea here for some later time.
